### ISSUE TITLE
Minecarts don't smoothly transition through portals

### ISSUE THREAD

**Issue by NikkeKatski** [qouteall/ImmersivePortalsMod#681]
I have a feeling that this is more of an issue with how Minecraft handles moving minecarts but it'd be sick if you could fix this somehow.

**Comment by gear54rus** [comment]
@qouteall  Why is this marked low prio? This seems like it could be another defining feature of this mod. Vanilla notoriously doesn't allow minecarts with entities to pass for example and fixing that in IP could enable smooth nether-enabled fast transportation networks (like the one I'm building :).

**Comment by gear54rus** [comment]
Or maybe you have some pointers on why it's hard to fix? like notes or something (relevant methods/classes). I could try to research and make a PR for that myself if given some hints (I never wrote any Java in my life but did modify several Java programs on bytecode level).

I can see there's MC 67 https://bugs.mojang.com/browse/MC-67 that's been open 12 years now lol. So I assume some kind of huge modification to portal code is needed but I can't see how it's so difficult if non-mounted entities (empty or storage minecarts) can already pass no problem.

**Comment by qouteall** [comment]
@gear54rus It involves position synchronization and interpolation of entities which is not easy to solve. I haven't looked into this issue deeply. Contributions are appreciated.

**Comment by qouteall** [comment]
In MC, the client player's movement is mainly determined by client side and synchronized to server side (server side validates the movement and send teleport packet to client if invalid). The position of non-client-player entities are determined by server side and is interpolated. So making non-client-player-entity to smoothly transition through portal requires different techniques than client player.

**Comment by gear54rus** [comment]
Makes sense. I can see how player entities might be an issue, but for example mobs would not have this problem right? I assume they are controlled by server.

If a minecart can pass through portal OK and a sheep can do the same, then both of them together should be able to? Or are there similar hidden issues?

**Comment by qouteall** [comment]
I think this issue is for visual smoothness of non-client-player teleporting (the visual smoothness issue applies for both minecart and non-minecart entities). You seem to mean the issue that minecart with passengers not teleporting through portal. They are two different issues.

**Comment by gear54rus** [comment]
oh yeah... I just realized I started this discussion in the wrong issue haha

**Comment by quulah** [comment]
Now that the support is apparently in with the vanilla portals in 1.21, is it possible to make this work with Immersive Portals?

I tried and the cart kind of goes half way through and gets stuck. No weird visual glitching however. :)
